code/app/src/main/java/com/github/bgabriel998/softwaredevproject/Button1Activity.java

```diff
@@ -1,14 +1,54 @@
 package com.github.bgabriel998.softwaredevproject;
 
+import androidx.annotation.NonNull;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.camera.core.CameraSelector;
+import androidx.camera.core.ImageAnalysis;
+import androidx.camera.core.ImageProxy;
+import androidx.camera.core.Preview;
+import androidx.camera.lifecycle.ProcessCameraProvider;
+import androidx.camera.view.PreviewView;
+import androidx.core.content.ContextCompat;
+import androidx.lifecycle.LifecycleOwner;
 
 import android.os.Bundle;
+import android.util.Size;
+
+import com.google.common.util.concurrent.ListenableFuture;
+
 
 public class Button1Activity extends AppCompatActivity {
 
+    private PreviewView previewView;
+    private ListenableFuture<ProcessCameraProvider> cameraProviderFuture;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_button1);
+
+        previewView = findViewById(R.id.view_finder);
+        cameraProviderFuture = ProcessCameraProvider.getInstance(this);
+        cameraProviderFuture.addListener(() -> {
+            try {
+                ProcessCameraProvider cameraProvider = cameraProviderFuture.get();
+                bindImageAnalysis(cameraProvider);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }, ContextCompat.getMainExecutor(this));
```

Comment:
Would be nice with some comment on what this part of the code does. I am a bit unsure on what to review since I don't really understand this.